tests/app/integration/test_pull_request.py

```diff
@@ -37,6 +37,62 @@ def test_invalid_policy():
     assert report.result == Result.FAIL, report.reason
 
 
+@pytest.mark.parametrize(
+    "github_branch, protected_github_branch, policy_enabled, expected_result",
+    [
+        pytest.param(
+            f"test-branch/pull_request/pass/{uuid4()}",
+            BranchWithProtection(),
+            True,
+            Result.FAIL,
+            id="disallow fork",
+        ),
+        pytest.param(
+            f"test-branch/pull_request/pass/{uuid4()}",
+            BranchWithProtection(),
+            False,
+            Result.PASS,
+            id="allow fork",
+        ),
+    ],
+    indirect=["github_branch", "protected_github_branch"],
+)
+@pytest.mark.usefixtures("protected_github_branch")
+def test_pull_request_disallow_fork(
+    github_branch: Branch,
+    github_repository_name: str,
+    forked_github_repository: Repository,
+    policy_enabled: bool,
+    expected_result: Result,
+):
+    """
+    arrange: given a source and target branch and repository that is not compliant
+    act: when pull_request is called
+    assert: then a fail report is returned.
```

Comment:
I think this is not quite correct as you also have a test which passes.

Reply:
Updated, thanks for catching that!